src/main/scala/workflow/WorkflowUtils.scala

```diff
@@ -0,0 +1,124 @@
+package workflow
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object WorkflowUtils {
+  def instructionsToPipeline[A, B](instructions: Seq[Instruction]): Pipeline[A, B] = {
+    val nodes = new ArrayBuffer[Node]()
+    val dataDeps = new ArrayBuffer[Seq[Int]]()
+    val fitDeps = new ArrayBuffer[Option[Int]]()
```

Comment:
I could replace this with a `foldLeft` and a case class w/ a merge method. If you prefer that, it might tighten the code up somewhat (up to 9 lines in this method or so, then it would add maybe 3 - 5 lines elsewhere), and it would remove the mutability.